### ISSUE TITLE
Suggestions/Cosmetic Enhancements & Bugs

### ISSUE THREAD

**Issue by ghost** [Mine-and-blade-admin/Battlegear2#128]
1. Melee/Unarmed Without Shield & General Issues
   Bug 0: The GUI is bugged out and seems to not work properly  . The button is clicked once but can't be unclicked .
   Fix Suggestion: ????

Bug 1: Enabling Battlegear mode "R" will always show offhand.  
Fix Suggestion: Leaving offhand empty should not render offhand. 

Bug 2: Right clicking to talk to someone while in Battlegear mode "R" will attack then with offhand once then enable them with right hand. 
Fix Suggestion: Same as above. Disable offhand if no weapon is placed in offhand. This is okay (for me) when dual wielding and I don't mind that at all. Having too many weapons reduces dexterity in the world making you hit friendlies or not able to mine or harvest certain things unless you place the weapon away. Makes sense. 

Minor Bug 3: Offhand seems to be permanently clickable unlike right hand which must be clicked over and over manually. 
Fix Suggestions: Make either hand behave like the other. Either make offhand have to manually click over and over like default hand OR make default hand permanently clickable. 

Bug 4: Looking down at the ground and holding and pressing offhand will make right/default hand flicker on and off. 
Fix Suggestion: There is probably an issue with the hands . The code may need re-writing. Easiest quick and dirty fix is simply disable continous clicking of offhand. Will still flicker hand but much less. 
1. Bow & Arrow:
   Bug 1: Placing bow in Offhand (L Hand) will work fine but arrow shoots from right hand and right hand is still visible.
   Bug 2: Placing bow in hand and arrow in offhand shows arrow as equipped in offhand. 
   Bug 3: Equipping bow in hand will still have right hand showing in 1st person. 
   Fix Suggestion: Bow is supposed to be 2 handed. When bow is equipped do not render offhand at all OR do not render offhand if any arrow or quiver are placed in offhand.  First method probably the easiest fix.

Bug 4: When choosing a quiver slot without any arrow slots and having arrows in inventory the bow will shoot out the inventory arrows in order but will not render the arrow in 1st person.
Fix Suggestion: If Player had arrows in inventory > 0 always render default arrow for bow and only render unique arrows when they are selected in quiver OR always render the arrow type based on the first arrow stack in inventory and if no arrows in inventory don't render any arrows. 

Major Bug 5: Disabling quiver render will crash minecraft when trying to position of the Battlegear gui. 
Fix Suggestion: ???

Major Bug 6: Disabling items in config file will crash minecraft when trying to load the Battlegear items tab in creative mode. The reason is null pointer. Somehow an item is disabled from the game but not from the gui. 
Fix Suggestion: ???
1. Shields
   Bug 1: Shield raising stance will not be visible in 3rd person.
   Fix Suggestion: Add the animations from 1st person raising offhand arm when shield is loaded into 3rd person. 

WEAPON SUGGESTION
Flail which you can stick a spiky weapon head and when moving left and right will drag the monster left and right. Flail will have spear range . See if you could collab with Balkon and mix the two mods.

MOD SUGGESTION
Redo the menu by moving the inventory crafting facing down and adding the extra weapon slots to the right. See for example something like this except mirrored left right http://img.photobucket.com/albums/v286/sakiruyonaki/inventoryexslots.png
I would modify the original menu instead of adding a new menu. 

Instead of using right click for left hand and left click for right hand you could use left click for both hands if both are equipped for left hand only if only left hand equipped or for right hand only if right hand equipped. This will make it much simpler. They attack at the same time but deal double damage due to having 2 weapons. 

Dual wielding should have a damage penalty: -1 on main hand and -2 on offhand from normal weapon damage.  This will still be crazy for example 2 iron swords 6.5 each that's 13 damage but - 3 that's 10 damage. Is more fair. Better than one sword, but not as good as each sword by itself. 

TO SUM UP
Easiest 2 fixes that will solve most bugs in here are
1. Disable offhand unless item is equipped & disable offhand completely when bows are equipped.  This will disable accidentally clicking on friendlies. 
2. Always render default arrow if player has 1 or more arrows in inventory. This will disable most bow and arrow bugs.
3. Disable continuous offhand clicking having to manually press left and right buttons manually. This will fix flickering issue

Thanks for the mod
Is really great . Even with the minor annoyances I find it very fun.
